pennylane/vqe/vqe.py

```diff
@@ -120,6 +120,52 @@ def __str__(self):
 
         return "\n+ ".join(terms)
 
+    def map_wires(self, wire_map):
+        r""" Maps the wires of a Hamiltonian to a new set of wires
+
+        Args:
+           wire_map (dict) : A dictionary that assigns new wires to a subset of the wires tracked in the Hamiltonian
+
+        Returns:
+            ~.Hamiltonian:
+        """
+
+        obs = {
+            "Identity": qml.Identity,
+            "PauliX": qml.PauliX,
+            "PauliY": qml.PauliY,
+            "PauliZ": qml.PauliZ,
+            "Hadamard": qml.Hadamard,
+            "Hermitian": qml.Hermitian,
+        }
+
+        if not isinstance(wire_map, dict):
+            raise ValueError(
+                "wire_map must be of type dict, got {}".format(type(wire_map).__name__)
+            )
+
+        f = lambda wire: wire_map[wire] if wire in wire_map.keys() else wire
+        new_terms = []
+
+        for term in self.ops:
+
+            tensor_term = []
+            term = qml.operation.Tensor(term) if isinstance(term.name, str) else term
+
+            for i in term.obs:
+
+                name = i.name
+                new_wires = list(map(f, i.wires))
+
+                if name == "Hermitian":
+                    tensor_term.append(obs[name](i.matrix, wires=new_wires))
+                else:
+                    tensor_term.append(obs[name](wires=new_wires))
```

Comment:
Here, you could instead do
```suggestion
                if name == "Hermitian":
                    tensor_term.append(qml.Hermitian(i.matrix, wires=new_wires))
                else:
                    tensor_term.append(i.__class__(wires=new_wires))
```

This avoids the need to create the `obs` dictionary, which might be hard to maintain if the observables change in other parts of the library.

Reply:
In fact, you can probably also drop the if statement:

```suggestion
                tensor_term.append(i.__class__(*i.data, wires=new_wires))
```